### DISCUSSION TITLE
Asynchronous search-bar problems!?

### DISCUSSION THREAD

**Discussion by RolandHesse** [neuefische/questions#788] [Web]
Hey there,

Unfortunately, our search-bar is still driving us crazy... What we want our search bar to do is the following:

- As soon as the user clicks into the search bar, a dropdown menu with the latest search terms is opened (a maximum of 5 of the latest search terms should be shown).
- As soon as the user starts typing into the search bar, the search begins and the list of displayed snippets is immediately updated so as to render only those snippets congruent with the entered search term.
- As soon as the user starts typing into the search bar, the list of last search terms is also updated so that only those last search terms that match the current search term are displayed.
- As soon as the user hits enter or clicks outside of the she search bar and/or the dropdown menu, the last search term array is updated. The list of last search terms in the dropdown menu is updated accordingly.
- As soon as the user clicks outside of the she search bar and/or the dropdown menu, the dropdown menu closes.
- As soon as the users clicks on one of the last search term terms, that term is displayed in the in search bar and the search is triggered.

Now, all of the above works - except for the last point: In order to trigger the desired behavior, the user has to click the pertinent last search term twice. At least this is so at the beginning (i.e., after a complete re-render of the site); after that, it works perfectly.

A second problem: Search results are not independent of what has been searched before. E.g., if you enter the letter "t" consecutively, the number of search results is diminished each time, until, when you have entered six "t"s ("tttttt") there is no matching results anymore. However, if you now start deleting "t"s, results only reappear when you deleted three "t"s. In short, in the "writing direction" results are shown until "ttttt" inclusively; in the "deleting direction", however, results only reappear at "ttt". 

We could imagine that both problems are connected to the asynchronous behaviour of state in React. In any case, we were unable to solve this issues.

As always: Any feedback is more than welcome! 🙏 

A PR for the latest version can be found here: https://github.com/PeetThesing/capstone-project_snippet-manager/pull/50

**Comment by doemser** [comment]
This ticket literally explodes. Please take a step back and review the current user-story and align your efforts with the originally agreed-upon objectives.

Supporting this ticket exceeds the feasible at this point.

I already created a prototype for you that handles almost every AC you described in this discussions thread. Please settle down for a version of the search that works and define a follow-up user-story for any additional features you want it to handle.
